sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -274,6 +274,8 @@ private[hive] class HiveClientImpl(
     val original = Thread.currentThread().getContextClassLoader
     // Set the thread local metastore client to the client associated with this HiveClientImpl.
     Hive.set(client)
+    // replace conf in cached hive with current conf
```

Comment:
@viirya `Hive.get(conf)` replaces conf in the **thread local** Hive with current conf, so it must be called after  `Hive.set(client)` which sets the current thread local Hive.